internal/backend/health/healthreporter/reporter.go

```diff
@@ -0,0 +1,5 @@
+package healthreporter
+
+type HealthReporter interface {
```

Comment:
I think a better way might be that instead an interface it will return a function with a specific type. Then the uber.Fx thingy can gather all of them and deliver them to the healthz endpoint, so it can iterate all of them.

see https://uber-go.github.io/fx/value-groups/

Reply:
this will require that we add additional code to each service we want to expose (fx.annotate to the health group)
I don't see how it makes it better / simpler

```   fx.Annotate(
      ServiceWithHealth,
      fx.As(new(HealthReporter)),
      fx.ResultTags(`group:"health_reporters"`),
    ),
```


if we plan to have loads of health reports, then I agree we should consider it
but for this use case, where we would probably have 2-3 (db/temporal/redis ? ) I think the explicit version is simpler and it also makes sure we know which health check are actually tested

WDYT?

Reply:
meh ok for now, but if we add more we'll def need to add it